streaming/text/convert/pile.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2022 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""English Wikipedia 2020-01-01 streaming dataset conversion script."""
+
+import json
+import os
+from argparse import ArgumentParser, Namespace
+from collections import Counter
+from glob import glob
+from multiprocessing import Pool
+from typing import Dict, Iterator, List, Tuple
+
+from streaming.base import MDSWriter
+from streaming.base.util import get_list_arg
+
+
+def parse_args() -> Namespace:
+    """Parse command-line arguments.
+
+    Args:
+        Namespace: command-line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument(
+        '--in_root',
+        type=str,
+        required=True,
+        help='Directory path to store the input dataset',
+    )
+    args.add_argument(
+        '--out_root',
+        type=str,
+        required=True,
+        help='Directory path to store the output dataset',
+    )
+    args.add_argument(
+        '--compression',
+        type=str,
+        default='zst:16',
+        help='Compression algorithm to use. Default: zst:16',
```

Comment:
How to know which compression algorithm to use ?

Reply:
- the user can judge the tradeoff for themself by running our compression benchmark on their own data
- compression levels seem to scale logarithmically in slowness
- i picked an aggressive level here because this dataset is quite large and this is a one-time calculation one on hand, and the highest number i could bear on the other hand

Reply:
Can we document this reasoning somewhere?